src/components/wallet/methods/depositAsset.ts

```diff
@@ -18,6 +19,11 @@ export default async function depositAsset(
     // If the asset exists in the wallet UI, there exists a record in this.assets
     // however, the homeDomain may not be defined if a trustline for that asset
     // wasn't added in the same session
+    // If there is no trustline established then we'll ask for the
+    // Assetcode and homedomain in order to make a Claimable Balance Deposit
+    if (!this.assets.get(`${asset_code}:${asset_issuer}`)) {
+      await getAssetAndIssuer(this)
+    }
```

Comment:
The primary reason was to ensure there was a homeDomain populated in order to initialize the deposit flow.
Without getting into too much in the weeds with the state management of wallet attributes; I can make this work as I think you intend it to via this code change 
```suggestion
    // If there is no trustline established then we'll ask for the
    // Assetcode and homedomain in order to make a Claimable Balance Deposit
    let homeDomain = null
    if (this.assets.get(`${asset_code}:${asset_issuer}`)){
      homeDomain = this.assets.get(`${asset_code}:${asset_issuer}`).homeDomain
    }
```
Note: This check needs to be made in order to not fail when attempting to access an attribute to a key value that isn't present (due to not saving the Wallet's asset property to disk). 

However another scenario that I think is an argument for keeping the `getAssetAndIssuer` call there is if you're debugging using an asset that doesn't have a home domain in the issuer's account.

For example, I issued a token on testnet named `LAWR` https://laboratory.stellar.org/#explorer?resource=assets&endpoint=single&values=eyJhc3NldF9jb2RlIjoiTEFXUiJ9&network=test
 However I did not set the [issuer's homedomain](https://developers.stellar.org/docs/start/list-of-operations/#set-options) primarily because it was anchored on a local debug instance of polaris.
 
 So I have 3 different paths to tackle this
 
 1. Make the change above and not allow anyone who hasn't set their homeDomain to make a deposit in the demo wallet.
 2. Keep the code the same and periodically (when you refresh) prompt the user to enter a homedomain to make a deposit.
 3. Refactor the code so we save the whole Wallet obj to disk (I believe I attempted this at one point and it didn't work out as intended but I can give it another go)

Reply:
Lets use your suggested change. If the home domain isn't set on the account, we ask for it on line 43.